### PR TITLE
Use single `GET` endpoints to retrieve GAT and PAT

### PR DESCRIPTION
**Pull request by timofurrer**
We should be able to merge this :)

### CODE REVIEW

**Comment by Shocktrooper** [comment]
@timofurrer Is there any GitHub action we can use that is similar to the !RemindMe 1 week/1 month/etc. of reddit. Where there can be a post/notification to look at a PR/issue when a certain period of time is reached? This would be useful for when functionality in the provider is blocked by a functionality that is scheduled to be released in a future GitLab version so as not to forget about something like this PR

**Comment by timofurrer** [comment]
@Shocktrooper a quick search revealed at least one: https://github.com/agrc/reminder-action

Though I'm not really sure if it's worth the maintenance - GitLab releases are always at the same day of the month, every month, thus it's kinda hard to miss - and I've created a milestone where to assign these issues / PRs. I think that gives us an additional benefit over the reminder, because we can actually filter them :)

**Comment by Shocktrooper** [comment]
@timofurrer The milestones seem like a good way to go.  I was just thinking for some issues where maybe the functionality is scheduled a long way down the line/not at all vs within a few months. They would be reminders to check if the upstream API has been released such as for an issue like [this](https://github.com/gitlabhq/terraform-provider-gitlab/issues/518) where GitLab has not added the functionality at the time and we might want to check in say like 6 months

**Comment by timofurrer** [comment]
This would break compatibility to GitLab < 14.10

**Comment by Shocktrooper** [comment]
@timofurrer Is this unblocked now that 14.10 has been released and the prerequisite PR has been merged?

File: internal/provider/resource_gitlab_group_access_token.go [hunk 1]
```diff
@@ -166,50 +166,31 @@ func resourceGitlabGroupAccessTokenRead(ctx context.Context, d *schema.ResourceD
...
-		if err != nil {
-			return diag.FromErr(err)
+	groupAccessToken, _, err := client.GroupAccessTokens.GetGroupAccessToken(group, groupAccessTokenId, gitlab.WithContext(ctx))
+	if err != nil {
+		if is404(err) {
+			log.Printf("[DEBUG] GitLab GroupAccessToken %d, group ID %s not found, removing from state", groupAccessTokenId, group)
+			d.SetId("")
+			return nil
 		}
+	}
```
Comment (PatrickRice-KSC) [inline internal/provider/resource_gitlab_group_access_token.go:176]:
```suggestion
	if err != nil && is404(err) {
		log.Printf("[DEBUG] GitLab GroupAccessToken %d, group ID %s not found, removing from state", groupAccessTokenId, group)
		d.SetId("")
		return nil
	}
```

Collapse nested IF statements :)

**Review by PatrickRice-KSC — CHANGES_REQUESTED** [review]
Losing internet on the flight since I'm landing, but one comment so far 😄

File: internal/provider/resource_gitlab_group_access_token.go (hunk 1 above)
Comment (timofurrer) [reply internal/provider/resource_gitlab_group_access_token.go]: This was actually a bug :D It should return an error if `err and not a 404` ...
I've updated the change set :)

**Comment by PatrickRice-KSC** [comment]
No other comments from reviews, thanks @timofurrer  and @Shocktrooper !

**Review by PatrickRice-KSC — APPROVED** [review]

**Pull request merged**